### PR TITLE
Add spell checker as Makefile target.

### PR DESCRIPTION
**Pull request by apahim**
This patch adds the `make spell` command so developers can easily
check the spelling on docstrings and comments. False positives
words should be added to spell.ignore file.

Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

**Comment by lmr** [comment]
Nice.

File: spell.ignore [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+postprocess
+gmail
+yaml
+tty
+ete
+http
+pythonhosted
+phylogenetics
+envs
+throuh
```
Comment (ldoktor) [inline spell.ignore]: This is suppose to be `through` in `./avocado/core/tree.py`

File: spell.ignore [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+distros
+logf
+gzipped
+basedir
+profiler
+mbligh
+apw
+shadowen
+mytest
+aaa
```
Comment (ldoktor) [inline spell.ignore]: This could be `foo/bar/baz` in `avocado/core/test.py`

File: spell.ignore [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+ci
+dtkit
+junit
+src
+thalesgroup
+xsd
+testsuite
+timestamp
+testcase
+rogram
```
Comment (ldoktor) [inline spell.ignore]: This is most probably suppose to be `program` in `avocado/core/remote/result.py`

File: spell.ignore [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+ldoktor
+msg
+zipfile
+tarfile
+symlinks
+xz
+rw
+hashfile
+stdlib
+englobing
```
Comment (ldoktor) [inline spell.ignore]: This one is located in `avocado/utils/astring.py` and is indeed a valid word, but I can't really understand, what it means. (even from the whole description, I only know I should escape something...)

File: spell.ignore [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+www
+tldp
+ldp
```
Comment (ldoktor) [inline spell.ignore]: I can't find this in avocado, where is it?

File: spell.ignore [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+nxm
+unix
+strems
```
Comment (ldoktor) [inline spell.ignore]: This is my typo in `avocado/utils/astring.py` whould be `streams`

File: spell.ignore [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+eg
+gb
+numa
+num
+vmstat
+smaps
+pid
+fragement
+buddyinfo
+splited
```
Comment (ldoktor) [inline spell.ignore]: This is probably suppose to be `splitted` in `avocado/utils/memory.py`

Comment (ldoktor) [inline spell.ignore]: or even just `split`? I'm not sure...

Comment (apahim) [inline spell.ignore]: Yep... `split`.

File: spell.ignore (hunk 5 above)
Comment (apahim) [inline spell.ignore]: It's not. Not sure why `pylint` got this. I will remove in v2.

File: spell.ignore (hunk 2 above)
Comment (apahim) [inline spell.ignore]: @ldoktor, this is the comment:
~~~
 93         # 001-mytest;aaa
 94         # 001-mytest;a
 95         # 001-myte;aaa
~~~
How do you suggest it to be?

File: spell.ignore (hunk 5 above)
Comment (apahim) [inline spell.ignore]: Oh, it's in `avocado/utils/astring.py`. Actually it's `LDP`, but `pylint` stores everything in lowercase,

File: spell.ignore (hunk 4 above)
Comment (apahim) [inline spell.ignore]: Well, the word is correct.. I understand the comment... The mentioned 'englobing' double quotes are the double quotes that delimit a string for bash.

File: spell.ignore (hunk 2 above)
Comment (ldoktor) [inline spell.ignore]: 001-mytest;foo
001-mytest;f
001-myte;foo

File: spell.ignore (hunk 5 above)
Comment (ldoktor) [inline spell.ignore]: OK, thanks for clarification (yes, it's understandable)

File: spell.ignore [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+ppid
+subshell
+prepended
+sigkill
+rc
+prerun
+expr
+rwx
+rx
+contet
```
Comment (ldoktor) [inline spell.ignore]: content `avocado/utils/script.py`

File: spell.ignore [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+unhashable
+safeloader
+loggables
+kosme
+noncharacters
+repr
+len
+hw
+scsi
+asci
```
Comment (ldoktor) [inline spell.ignore]: This should be ascii

File: spell.ignore [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+stacktrace
+exps
+umask
+hardcode
+decompressdir
+usa
```
Comment (ldoktor) [inline spell.ignore]: Nice USA is not on the list :smile:

File: spell.ignore [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+si
+sourceware
+harisrir
+jadmanski
+ba
+ag
+meth
+arcname
+escapingsection
+asdf
```
Comment (ldoktor) [inline spell.ignore]: Oh, me to blame again. It's probably suppose to be foo/bar/baz, what do you think? Or is it acceptable?

File: spell.ignore [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+aleax
+prepend
+behaviour
+hudson
+jhcepas
+raphtee
+fdsa
```
Comment (ldoktor) [inline spell.ignore]: The same as `asdf`

**Comment by ldoktor** [comment]
Oh, I'm sorry I had to interrupt the review yesterday and added few comments here below the already fixed ones (last third of the commit). Can you please include the changes in v2?

**Comment by apahim** [comment]
Sure @ldoktor. no problem. v2 updated.

**Pull request closed without merging**